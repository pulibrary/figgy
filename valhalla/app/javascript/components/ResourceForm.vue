<template>
  <div>
    <h2>Set Properties <small>for this resource</small></h2>
    <div class="row">
      <!-- Viewing Direction-->
      <div class="form-group col-md-6">
        <fieldset>
          <legend>Viewing Direction</legend>
          <div v-for="viewDir in viewDirs">
            <div class="radio">
              <label>
                <input class="viewDirInput" @change="updateViewDir($event)" v-model="viewingDirection" type="radio" name="viewDir" :value="viewDir.label">
                {{viewDir.label}}
              </label>
            </div>
          </div>
        </fieldset>
      </div>
      <!-- Viewing Hint-->
      <div class="form-group col-md-6">
        <fieldset>
          <legend>Viewing Hint</legend>
          <div v-for="viewHint in viewHints">
            <div class="radio">
              <label>
                <input class="viewHintInput" @change="updateViewHint($event)" v-model="viewingHint" type="radio" name="viewHint" :value="viewHint.label">
                {{viewHint.label}}
              </label>
            </div>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'resource-form',
  data: function () {
    return {
      viewDirs: [
        {label: "left-to-right"},
        {label: "right-to-left"},
        {label: "top-to-bottom"},
        {label: "bottom-to-top"},
      ],
      viewHints: [
        {label: "individuals"},
        {label: "paged"},
        {label: "continuous"}
      ]
    }
  },
  computed: {
    viewingDirection: function () {
      return this.$store.state.viewingDirection
    },
    viewingHint: function () {
      return this.$store.state.viewingHint
    }
  },
  methods: {
    updateViewDir (event) {
      var value = event.target.value
      this.$store.dispatch('updateViewDir', value)
    },
    updateViewHint (event) {
      var value = event.target.value
      this.$store.dispatch('updateViewHint', value)
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

fieldset {
  margin: auto;
}

legend {
  font-size: 14px;
  margin: auto;
}

.radio label {
  margin: auto;
  padding-left: 30px;
}

</style>
