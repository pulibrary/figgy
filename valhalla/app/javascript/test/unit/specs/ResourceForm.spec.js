import Vuex from 'vuex'
import { mount, createLocalVue } from 'vue-test-utils'
import ResourceForm from '@/components/ResourceForm'
const localVue = createLocalVue()
localVue.use(Vuex)

describe('ResourceForm.vue', () => {
  let wrapper
  let options
  let actions
  let state
  let store

  beforeEach(() => {
    actions = {
      updateViewDir: jest.fn(),
      updateViewHint: jest.fn()
    }
    state = {
      viewingDirection: 'bottom-to-top',
      viewingHint: 'continuous'
    }
    store = new Vuex.Store({
      state,
      actions
    })

    options = {
      computed: {
        viewingDirection: function () {
          return 'bottom-to-top'
        },
        viewingHint: function () {
          return 'continuous'
        }
      }
    }
  })

  // it('renders a $store.state value return from computed', () => {
  //   expect(cmp.find('#detail_img').html()).toEqual('<img id="detail_img" src="http://localhost:3000/image-service/50b5e49b-ade7-4278-8265-4f72081f26a5/full/400,/0/default.jpg">')
  // })

  it('allows the selection of a viewing direction', () => {
    const wrapper = mount(ResourceForm, { options, store, localVue })
    wrapper.find('.viewDirInput').trigger('change')
    expect(actions.updateViewDir).toHaveBeenCalled()
  })

  it('allows the selection of a viewing hint', () => {
    const wrapper = mount(ResourceForm, { options, store, localVue })
    wrapper.find('.viewHintInput').trigger('change')
    expect(actions.updateViewHint).toHaveBeenCalled()
  })

  it('has the expected html structure', () => {
    const wrapper = mount(ResourceForm, { options, store, localVue })
    expect(wrapper.element).toMatchSnapshot()
  })

})
