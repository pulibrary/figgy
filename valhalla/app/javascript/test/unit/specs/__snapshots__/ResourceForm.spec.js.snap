// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceForm.vue has the expected html structure 1`] = `
<div>
  <h2>
    Set Properties 
    <small>
      for this resource
    </small>
  </h2>
   
  <div
    class="row"
  >
    <div
      class="form-group col-md-6"
    >
      <fieldset>
        <legend>
          Viewing Direction
        </legend>
         
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewDirInput"
                name="viewDir"
                type="radio"
              />
              
              left-to-right
            
            </label>
          </div>
        </div>
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewDirInput"
                name="viewDir"
                type="radio"
              />
              
              right-to-left
            
            </label>
          </div>
        </div>
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewDirInput"
                name="viewDir"
                type="radio"
              />
              
              top-to-bottom
            
            </label>
          </div>
        </div>
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewDirInput"
                name="viewDir"
                type="radio"
              />
              
              bottom-to-top
            
            </label>
          </div>
        </div>
      </fieldset>
    </div>
     
    <div
      class="form-group col-md-6"
    >
      <fieldset>
        <legend>
          Viewing Hint
        </legend>
         
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewHintInput"
                name="viewHint"
                type="radio"
              />
              
              individuals
            
            </label>
          </div>
        </div>
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewHintInput"
                name="viewHint"
                type="radio"
              />
              
              paged
            
            </label>
          </div>
        </div>
        <div>
          <div
            class="radio"
          >
            <label>
              <input
                class="viewHintInput"
                name="viewHint"
                type="radio"
              />
              
              continuous
            
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;
